org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ClasspathCommand.java

```diff
@@ -0,0 +1,204 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.commands;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IClassFile;
+import org.eclipse.jdt.core.IClasspathContainer;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.core.PackageFragment;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+public class ClasspathCommand {
+
+	private static Map<String, Function<List<String>, Either<ClasspathItem[], String>>> commands = new HashMap<>();
+
+	static {
+		commands.put(ClasspathItem.CONTAINER, ClasspathCommand::getContainers);
+		commands.put(ClasspathItem.JAR, ClasspathCommand::getJars);
+		commands.put(ClasspathItem.PACKAGE, ClasspathCommand::getPackages);
+		commands.put(ClasspathItem.CLASSFILE, ClasspathCommand::getClassfiles);
+		commands.put(ClasspathItem.SOURCE, ClasspathCommand::getSource);
+	}
+
+	public static Either<ClasspathItem[], String> getClasspathItems(List<Object> arguments) throws CoreException {

```

Comment:
You need to add some Javadoc to describe what that method does and return

Reply:
Method should simply return ClasspathItem[], not content

Reply:
Why not just return a List<ClasspathItem>? Also, why encode args as a list when you can just read them from argumens(1..n) ?

Reply:
Fixed